geomagio/WebService.py

```diff
@@ -81,7 +78,77 @@ def parse(params):
         WebServiceQuery
             parsed query object.
         """
-        pass
+        # Create dictionary of lists
+        dict = parse_qs(params)
+        # Return values
+        id = dict.get('id', [''])[0]
+        starttime = dict.get('starttime', [''])[0]
+        endtime = dict.get('endtime', [''])[0]
+        elements = dict.get('elements', [''])[0]
+        sampling_period = dict.get('sampling_period', [''])[0]
+        type = dict.get('type', [''])[0]
+        format = dict.get('format', [''])[0]
+        # Escape to avoid script injection
+        self.id = escape(id)
+        self.starttime = escape(starttime)
+        self.endtime = escape(endtime)
+        elements = escape(elements)
+        self.sampling_period = escape(sampling_period)
+        self.type = escape(type)
+        self.format = escape(format)
+        # Check for values
+        if not self.id:
+            self.id = 'BOU'
+        if not starttime or not endtime:
+            self.starttime = str(datetime.utcnow().strftime("%Y-%m-%dT")) + '00:00:00Z'
+            self.endtime = str(datetime.utcnow().strftime("%Y-%m-%dT")) + '23:59:00Z'
+        if elements:
+            self.elements = [el.strip() and el.upper() for el in elements.split(',')]
+        else:
+            self.elements = ('X','Y','Z','F')
+        if self.sampling_period == '1':
+            self.sampling_period = 'second'
+        if self.sampling_period == '60':
+            self.sampling_period = 'minute'
+        # TODO: Add hourly option
+        if not self.sampling_period:
+            self.sampling_period = 'minute'
+        if not self.type:
+            self.type = 'variation'
+        if not self.format:
+             self.format = 'iaga2002'
+
+
+
+
+    @classmethod
```

Comment:
`classmethod` works on the global class, `self` in this case refers to the class object rather than an instance, so concurrent requests would potentially have a race condition.